Sources/Utilities/ImageTools/ImageTools.swift

```diff
@@ -0,0 +1,358 @@
+//
+//  ImageTools.swift
+//  PovioKit
+//
+//  Created by Borut Tomazin on 13/06/2024.
+//  Copyright © 2024 Povio Inc. All rights reserved.
+//
+
+import UIKit
+import PovioKitCore
+
+/// A utility class for performing various image-related operations.
+public final class ImageTools: NSObject {
+#if canImport(Kingfisher)
+  private var prefetcher: ImagePrefetcher?
+#endif
+  
+  override public init() {
+    super.init()
+  }
+}
+
+public extension ImageTools {
+  enum ImageError: LocalizedError {
+    case invalidPercentage
+    case compression
+    case invalidSize
+  }
+  
+  enum ImageFormat {
+    case jpeg(compressionRatio: CGFloat)
+    case png
+  }
+  
+  /// Saves the given image to the photo library asynchronously.
+  ///
+  /// This function saves the provided UIImage to the user's photo library. It performs
+  /// the operation asynchronously and throws an error if the save operation fails.
+  ///
+  /// - Parameters:
+  ///   - image: The UIImage to be saved to the photo library.
+  /// - Example:
+  /// ```swift
+  /// do {
+  ///   let image = UIImage(named: "exampleImage")!
+  ///   try await saveImageToPhotoLibrary(image)
+  ///   print("Image saved successfully.")
+  /// } catch {
+  ///   print("Failed to save image: \(error)")
+  /// }
+  /// ```
+  /// - Note: This function should be called from an asynchronous context using `await`.
+  /// - Throws: An error if the save operation fails.
+  func saveToPhotoLibrary(image: UIImage) async throws {
+    try await withCheckedThrowingContinuation { continuation in
+      let continuationWrapper = ContinuationWrapper(continuation: continuation)
+      UIImageWriteToSavedPhotosAlbum(
+        image,
+        self,
+        #selector(saveCompleted),
+        Unmanaged.passRetained(continuationWrapper).toOpaque()
+      )
+    }
+  }
```

Comment:
Same here, I would leave it as synchronous.

Reply:
Cannot do that. We need to have an async code just to support old selector based api.